shared/common-adapters/input.native.js

```diff
@@ -233,31 +213,45 @@ class Input extends Component<Props, State> {
       multilineStyle.height = this.state.height
     }
 
+    const value = this.getValue()
+
     const floatingHintText =
-      !!this.state.value.length &&
+      !!value.length &&
       (this.props.hasOwnProperty('floatingHintTextOverride')
         ? this.props.floatingHintTextOverride
         : this.props.hintText || ' ')
 
+    let keyboardType: ?KeyboardType = this.props.keyboardType
+    if (!keyboardType) {
+      if (isAndroid && this.props.type === 'passwordVisible') {
+        keyboardType = 'visible-password'
```

Comment:
Tested this and confirmed that this does what we want (it was silently broken, per above).